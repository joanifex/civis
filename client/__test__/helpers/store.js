import configureMockStore from 'redux-mock-store';


const store = (model) => {
  const mockStore = configureMockStore();
  return mockStore(model)
}

export default store;
