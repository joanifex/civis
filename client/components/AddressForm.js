import React from 'react';
import { browserHistory } from 'react-router';

class AddressForm extends React.Component {

  handleSubmit = (e) => {
    e.preventDefault();
    let address = this.address.value
    this.updateAddress({address})
  }

  geolocate = () => {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition( position => {
        let coords = {
          lat: position.coords.latitude,
          lng: position.coords.longitude
        };
        this.updateAddress({coords})
      });
    } else {
      Materialize.toast("Can't Geolocate.")
    }
  }

  updateAddress = ({ address = "", coords = [] }) => {
    $.ajax({
      type: 'PATCH',
      url: `/api/user/address`,
      dataType: 'JSON',
      data: { address, coords }
    }).done(data => {
      Materialize.toast('Address Updated', 3000);
      if (this.props.addressEntered)
        this.props.addressEntered();
    }).fail( data => {
      Materialize.toast('Invalid address, Please try again', 3000);
    });
  }

  render(){
    return(
      <div>
        <span className='card-title center'>
          Find your legislators by zip code or address.
        </span>
        <form className='center' onSubmit={this.handleSubmit}>
          <div className="row">
            <input
             ref={ n => this.address = n}
             className="col s12 m8 offset-m1"
             placeholder='Zip code or Address'
             required
             autoFocus
            />
            <input className='btn blue-grey' type='submit'/>
          </div>
        </form>
        <button className="btn" onClick={this.geolocate}>Geolocate</button>
      </div>
    );
  }
}

export default AddressForm;
