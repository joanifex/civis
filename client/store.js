import { createStore, compose, applyMiddleware } from 'redux';
import { syncHistoryWithStore } from 'react-router-redux';
import { browserHistory } from 'react-router';
import thunk from 'redux-thunk';

import rootReducer from './reducers/index';

const enhancers = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)

let appDataSet = document.getElementById('app').dataset;
let remember = appDataSet.remember;

remember = remember === 'true' ? true : false;

const store = createStore(rootReducer, { auth: { isAuthenticated: remember } }, enhancers)

if(module.hot) {
  module.hot.accept('./reducers/',() => {
    const nextRootReducer = require('./reducers/index').default;
    store.replaceReducer(nextRootReducer);
  });
}

export const history = syncHistoryWithStore(browserHistory, store)

export default store
