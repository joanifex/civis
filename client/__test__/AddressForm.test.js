import React from 'react';
import AddressForm from '../components/AddressForm.js';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from './helpers/store';

test('AddressForm is rendered', () => {
  const component = renderer.create(
    <Provider store={store({})}>
      <AddressForm />
   </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
