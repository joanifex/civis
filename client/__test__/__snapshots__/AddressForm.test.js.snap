// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm is rendered 1`] = `
<div>
  <span
    className="card-title center"
  >
    Find your legislators by zip code or address.
  </span>
  <form
    className="center"
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <input
        autoFocus={true}
        className="col s12 m8 offset-m1"
        placeholder="Zip code or Address"
        required={true}
      />
      <input
        className="btn blue-grey"
        type="submit"
      />
    </div>
  </form>
</div>
`;
